lib/liquid/errors.rb

```diff
@@ -40,14 +43,19 @@ def message_prefix
     end
   end
 
+  class MemoryError < Error
+    RENDER_LENGTH_ERROR_MESSAGE = 'Render length limits exceeded.'
+    RENDER_SCORE_ERROR_MESSAGE = 'Render score limits exceeded.'
+    ASSIGN_SCORE_ERROR_MESSAGE = 'Assign score limits exceeded.'
```

Comment:
Maybe just wording like:

"too many bytes rendered"
"too many tags rendered"
"too many bytes assigned to variables"